### PR TITLE
UI home cards

### PR DESCRIPTION
**Pull request by emmaodia**
CTA implementation cc: @hachiojidev @Chef-Chungus 

[Netlify Status](https://app.netlify.com/sites/vibrant-noyce-fadd4c/deploys)

[x] Earn up to (%) APY - Must obtain the highest current apy offered on the site, and display it here.
[x] Earn (ASSET, ASSET, ASSET, ASSET) in Pools - select from list of active pools (available in config)
[x] Win up to ($) in Lottery Card
[x] Must convert CAKE to USD 

View here: https://vibrant-noyce-fadd4c.netlify.app/

Dark Mode
dark-mode

Light Mode
light-mode

Must convert CAKE to USD implemented
usdvalue-implemented

### CODE REVIEW

**Comment by hachiojidev** [comment]
@emmaodia  Is this the same as https://github.com/pancakeswap/pancake-frontend/pull/210?

**Comment by emmaodia** [comment]
> @emmaodia Is this the same as #210?

Yes. I don't have access to push to the head of 210. I forked the repo and this PR is across forks. @hachiojidev

File: src/views/Home/components/EarnAssetCard.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+`
+
+const EarnAssetCard = () => {
+  const TOKENS_TO_EXCLUDE = ['SYRUP']
+  const tokenNames = pools.reduce((accum, pool) => {
+    if (TOKENS_TO_EXCLUDE.includes(pool.tokenName)) {
+      return accum
+    }
+
+    return [...accum.slice(0, 3), pool.tokenName]
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAssetCard.tsx:38]: Why don't you just put the first 3 tokens in the exclude list?

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "baseUrl": "src",
-    "target": "es5",
+    "target": "es2017",
```
Comment (hachiojidev) [inline tsconfig.json:4]: Why was this changed?

File: tsconfig.json [hunk 3]
```diff
@@ -24,6 +24,6 @@
     "noFallthroughCasesInSwitch": true
   },
   "include": [
-    "src"
+    "./src/**/*.ts"
```
Comment (hachiojidev) [inline tsconfig.json:27]: Same...

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 3 above)
Comment (emmaodia) [reply tsconfig.json:27]: These happened when I ran yarn install after forking the project. It is an auto updated. 
I have reverted it.


tsconfig-1

File: src/views/Home/components/EarnAssetCard.tsx (hunk 1 above)
Comment (emmaodia) [reply src/views/Home/components/EarnAssetCard.tsx:38]: This line returns the first 4 tokens from the list of tokens. The card is meant to display 4 tokens according to the Figma file:
figma-assetsCard

**Comment by emmaodia** [comment]
Update: Fixed color display contrast for Earn Asset Card in Dark and Light modes:

dark-card

light-card

File: src/views/Home/components/EarnAPYCard.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  max-width: 344px;
+  width: 100%;
+
+  ${({ theme }) => theme.mediaQueries.lg} {
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  color: ${({ theme }) => theme.colors.contrast};
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAPYCard.tsx:22]: Heading accepts the color attribute. You can use it like this.

```
<Heading color="contrast" />
```

File: src/views/Home/components/EarnAPYCard.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
+  color: #7645d9;
+`
+const Row = styled.div`
+  display: flex;
+  justify-content: flex-end;
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAPYCard.tsx:29]: You can use the `<Flex />` component from the UI Kit.

File: src/views/Home/components/EarnAssetCard.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  color: ${({ theme }) => theme.colors.contrast};
+`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
+  color: ${({ theme }) => theme.colors.invertedContrast};
+  font-size: 35px;
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAssetCard.tsx:23]: Same as above.

Plus, please use the font sizes from the theme. Figma has them at 44px and 24px respectively.

File: src/views/Home/components/WinCard.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  color: ${({ theme }) => theme.colors.contrast};
+`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
+  color: #7645d9;
+`
```
Comment (hachiojidev) [inline src/views/Home/components/WinCard.tsx:25]: Heading also accepts hex values not in the theme.

File: src/views/Home/components/WinCard.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+const Row = styled.div`
+  display: flex;
+  justify-content: flex-end;
```
Comment (hachiojidev) [inline src/views/Home/components/WinCard.tsx:29]: `<Flex />`

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Home/components/EarnAssetCard.tsx (hunk 6 above)
Comment (emmaodia) [reply src/views/Home/components/EarnAssetCard.tsx:23]: font-size: 40px;
 line-height: 44px;

**Comment by emmaodia** [comment]
@hachiojidev  Requested changes made. Thank you.

arrow-primary

File: src/views/Home/components/EarnAPYCard.tsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  font-size: 24px;
+`
+const CardMidContent = styled(Heading)`
+  font-size: 40px;
+  line-height: 44px;
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAPYCard.tsx:26]: Please check the UI Kit. It has a size prop.

```
<Heading size="lg" /> // 24px
<Heading size="xl" /> // 40px
```

File: src/views/Home/components/EarnAssetCard.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  font-size: 24px;
+`
+const CardMidContent = styled(Heading)`
+  font-size: 40px;
+  line-height: 44px;
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAssetCard.tsx:23]: Same here

File: src/views/Home/components/WinCard.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    margin: 0;
+    max-width: none;
+  }
+`
+const Label = styled(Heading)`
+  font-size: 24px;
+`
+const CardMidContent = styled(Heading)`
+  font-size: 40px;
+  line-height: 44px;
```
Comment (hachiojidev) [inline src/views/Home/components/WinCard.tsx:25]: And here

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Home/components/EarnAPYCard.tsx [hunk 12]
```diff
@@ -18,10 +18,10 @@ const StyledFarmStakingCard = styled(Card)`
     max-width: none;
   }
 `
-const Label = styled(Heading)`
+const Label = styled(Heading).attrs({ size: 'lg' })`
   font-size: 24px;
 `
-const CardMidContent = styled(Heading)`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAPYCard.tsx]: The `size` prop determines the font size of the component. So you don't need this component but can use the Heading component directly

```
<Heading size="lg" />
```

File: src/views/Home/components/EarnAssetCard.tsx [hunk 13]
```diff
@@ -15,10 +15,10 @@ const StyledFarmStakingCard = styled(Card)`
...
 `
-const Label = styled(Heading)`
+const Label = styled(Heading).attrs({ size: 'lg' })`
   font-size: 24px;
 `
-const CardMidContent = styled(Heading)`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
   font-size: 40px;
   line-height: 44px;
 `
```
Comment (hachiojidev) [inline src/views/Home/components/EarnAssetCard.tsx]: This case you need to modify the line height so it works. Just remove the font-size.

File: src/views/Home/components/WinCard.tsx [hunk 14]
```diff
@@ -17,10 +17,10 @@ const StyledFarmStakingCard = styled(Card)`
...
 `
-const Label = styled(Heading)`
+const Label = styled(Heading).attrs({ size: 'lg' })`
   font-size: 24px;
 `
-const CardMidContent = styled(Heading)`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
   font-size: 40px;
   line-height: 44px;
 `
```
Comment (hachiojidev) [inline src/views/Home/components/WinCard.tsx]: Same as above...

**Review by hachiojidev — CHANGES_REQUESTED** [review]

**Comment by hachiojidev** [comment]
@emmaodia This card needs to show the highest APY.
image

**Comment by self-coding-crab** [comment]
We might specify types for everything

File: src/views/Home/components/EarnAPYCard.tsx [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+  }
+`
+const CardMidContent = styled(Heading).attrs({ size: 'xl' })`
+  line-height: 44px;
+`
+const EarnAPYCard = () => {
+  const TranslateString = useI18n()
+  const farmsLP = useFarms()
+
+  const calculateAPY = () => {
```
Comment (self-coding-crab) [inline src/views/Home/components/EarnAPYCard.tsx:28]: Define return type

**Review by hachiojidev — CHANGES_REQUESTED** [review]
2 things:

1. Can you update your branch. Changes have been made to the dashboard that are relevant.
2. Please add a loading state to the APY with the `<Skeleton />` component.

**Comment by emmaodia** [comment]
> 2 things:
> 
> 1. Can you update your branch. Changes have been made to the dashboard that are relevant.
> 2. Please add a loading state to the APY with the `<Skeleton />` component.

Done. https://6010639a78eb2109ba8c46b2--vibrant-noyce-fadd4c.netlify.app/

**Comment by hachiojidev** [comment]
@emmaodia One final thing. We noticed the lottery prize amount was incorrect. We are doing other work related to this so I added a hook to `develop`. Please use this and we are good to go.

https://github.com/pancakeswap/pancake-frontend/blob/develop/src/hooks/useLotteryTotalPrizesUsd.ts

Note - it returns the number in USD unformatted.

**Review by hachiojidev — APPROVED** [review]
